### PR TITLE
Fixed plugin failure on Issue with no priority.

### PR DESCRIPTION
**Pull request by DummyThatMatters**
#### Summary
Fixes the situation when plugin process the issue with Priority field set to `nil`, which can potentially lead to situation when nill pointer dereference error occurs.

#### Ticket Link
Seem to fix https://github.com/mattermost/mattermost-plugin-jira/issues/820

### CODE REVIEW

**Comment by mickmister** [comment]
Hi @DummyThatMatters, thanks for catching this and providing the fix :tada: 

Just so I can understand the issue better, do you know if this is occurring in one of these two cases:

- the issue scheme does not contain the `priority` field
- the issue scheme does have a `priority`, but the priority is not assigned for a given issue somehow

As far as the fix you've provided, I have one request to provide the same fix to other fields potentially having this issue in the same function. Thanks for this great contribution @DummyThatMatters!

**Review by mickmister — CHANGES_REQUESTED** [review]

File: server/issue.go [hunk 1]
```diff
@@ -785,7 +785,9 @@ func getIssueFieldValue(issue *jira.Issue, key string) StringSet {
 	case labelsField:
 		return NewStringSet(issue.Fields.Labels...)
 	case priorityField:
-		return NewStringSet(issue.Fields.Priority.ID)
+		if issue.Fields.Priority != nil {
+			return NewStringSet(issue.Fields.Priority.ID)
+		}
```
[the remark this answers is not in the document]
Comment (DummyThatMatters) [reply server/issue.go:790]: > Hi @DummyThatMatters, thanks for catching this and providing the fix 🎉
> 
> Just so I can understand the issue better, do you know if this is occurring in one of these two cases:
> 
>     * the issue scheme does not contain the `priority` field
> 
>     * the issue scheme does have a `priority`, but the priority is not assigned for a given issue somehow
> 
> 
> As far as the fix you've provided, I have one request to provide the same fix to other fields potentially having this issue in the same function. Thanks for this great contribution @DummyThatMatters!

Frankly, I don't really know. Does this thing matter? We debugged this a long time ago and I have implemented my patch on our prod server, so it no longer fail and I don't know which exact tickets caused the issue. 
I can try to find out if this info is essential in order to investigate the bug and fix potential problems. And we do not have any problems with other fields for some reason.

[the remark this answers is not in the document]
Comment (mickmister) [reply server/issue.go:790]: @DummyThatMatters Fair enough, I'll approve at its current state. Note that you'll need to sign our CLA before getting this merged, as explained here https://github.com/mattermost/mattermost-plugin-jira/pull/821#issuecomment-970582277

Thanks again for this fix 😄

**Review by mickmister — APPROVED** [review]

**Comment by dipak-demansol** [comment]
@DummyThatMatters priority is a mandatory field at the time of creating jira issue using the plugin so how you reported the issue with No priority?

**Comment by dipak-demansol** [comment]
> @DummyThatMatters priority is a mandatory field at the time of creating jira issue using the plugin so how you reported the issue with No priority?

@mickmister can you give me some info to regenerate the issue?

**Comment by DummyThatMatters** [comment]
> @DummyThatMatters priority is a mandatory field at the time of creating jira issue using the plugin so how you reported the issue with No priority?

@dipak-demansol hello! You can track issues that have not been created via plugin. then you'll have some problems with nil priority issues.

**Comment by mickmister** [comment]
@dipak-demansol I think you'll need to have an issue schema that has a priority field, but no default value set for the field. Not entirely sure how that works though

**Comment by dipak-demansol** [comment]
@DHaussermann We're trying to find a way to null out priority from web request.

**Review by hanzei — APPROVED** [review]

**Comment by hanzei** [comment]
@DHaussermann Can we merge this straight forward fix would QA review?

**Comment by hanzei** [comment]
@DummyThatMatters Would you be open on signing the CLA?

**Comment by DHaussermann** [comment]
@hanzei If this is low risk - we should merge it without testing to resolve the problem. 

0/5 There may have simple ways to build the data needed for this in the past. But i believe this filed is automatically set even when removed from all configuration and not visible in the UI. (Jira seems to assign values which it uses to hold the order of tings in lists.)

In any case, I can find no simple way to create the data needed short of manipulating a DB. I suggest we merge this and rely on release testing to catch any regression.

**Review by DHaussermann — APPROVED** [review]
Please disregard previous comment. I had the filed and scenario confused with something else. 

Tested and passed.
Steps:
- Picked a project  
- Created a custom scheme for issues of type Bug within the ptoject
- Removed priority field from the scheme
- Confirmed the Jira webapp  could create the issue
- Confirmed the Jira plugin can create the same issue form within Mattermost

This issue is resolved. No regressions found

LGTM!

Thanks for the soulution @DummyThatMatters 🎉

**Pull request merged**